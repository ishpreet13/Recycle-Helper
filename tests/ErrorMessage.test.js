import "react-native";
import React from "react";
import renderer from "react-test-renderer";

import ErrorMessage from "../app/components/ErrorMessage";

describe("Testing Error Message Component when it is visible and error is present", () => {
  const wrapper = renderer.create(
    <ErrorMessage visible={true} error={"Fake Error"} />
  );
  it("should render", () => {
    expect(wrapper.toJSON()).toBeTruthy();
  });
});

describe("Testing Error Message Component when it is not visible and error is present", () => {
  const wrapper = renderer.create(
    <ErrorMessage visible={false} error={"Fake Error"} />
  );
  it("should not render", () => {
    expect(wrapper.toJSON()).toBeNull();
  });
});

describe("Testing Error Message Component when it is  visible and error is not present", () => {
  const wrapper = renderer.create(<ErrorMessage visible={true} error={""} />);
  it("should not render", () => {
    expect(wrapper.toJSON()).toBeNull();
  });
});

describe("Testing Error Message Component when it is not visible and error is not present", () => {
  const wrapper = renderer.create(<ErrorMessage visible={false} error={""} />);
  it("should not render", () => {
    expect(wrapper.toJSON()).toBeNull();
  });
});

describe("<ErrorMessage />", () => {
  let tree;
  it("should render correctly", async () => {
    tree = renderer.create(
      <ErrorMessage visible={true} error={"Fake Error"} />
    );
    expect(tree.toJSON()).toMatchSnapshot();
  });
});
