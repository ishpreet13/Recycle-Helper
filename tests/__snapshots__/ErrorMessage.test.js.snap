// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ErrorMessage /> should render correctly 1`] = `
<Text
  style={
    Object {
      "color": "#000",
      "fontSize": 18,
      "fontWeight": "600",
      "marginBottom": 10,
    }
  }
>
  ⚠️ 
  Fake Error
</Text>
`;
